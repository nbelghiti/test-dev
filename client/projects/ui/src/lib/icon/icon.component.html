<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
