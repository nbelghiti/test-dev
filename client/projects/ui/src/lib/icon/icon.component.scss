// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
