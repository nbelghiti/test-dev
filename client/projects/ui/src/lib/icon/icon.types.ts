export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
