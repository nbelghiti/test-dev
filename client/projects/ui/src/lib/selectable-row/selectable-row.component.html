<label class="c-selectable-row">
	<input
		class="c-selectable-row__checkbox"
		type="radio"
		[name]="name"
		[value]="value"
		[formControl]="isChecked"
	/>

	<div
		class="c-selectable-row__container"
	>
		<div class="c-selectable-row__content">
			<div
				*ngIf="iconBefore"
				class="c-selectable-row__icon c-selectable-row__icon--before"
			>
				<ui-icon [name]="iconBefore"></ui-icon>
			</div>

			<ng-content></ng-content>
		</div>
	</div>
</label>
