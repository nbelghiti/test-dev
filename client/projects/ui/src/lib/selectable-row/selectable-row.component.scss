// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Selectable Row
// =============================================================================
.c-selectable-row {
	width: 100%;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-selectable-row__container {
	position: relative;

	display: flex;
	justify-content: space-between;
	align-items: center;
	width: 100%;
	height: 100%;
	min-height: 5.2rem;
	overflow: hidden;
	padding: 0.5rem 1.6rem;

	text-decoration: none;

	border: $border-width-base solid $border-color-base;
	background-color: $background-color-base;
	color: inherit;
	box-shadow: none;

	transition:
		border-color $animation-base,
		background-color $animation-base;

	&:hover {
		border-color: $primary-color;

		cursor: pointer;
	}
}

.c-selectable-row__content {
	display: flex;
	align-items: center;
	flex-grow: 1;
}

.c-selectable-row__checkbox {
	position: absolute;

	width: 0;
	height: 0;

	opacity: 0;

	&:checked {

		+ .c-selectable-row__container {
			border-color: $primary-color;
			background-color: rgba($primary-color, 0.08);
		}
	}
}

.c-selectable-row__icon {
	display: flex;

	font-size: 1.5rem;
	line-height: $line-height-base;

	color: currentColor;

	transition: color $animation-base;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-selectable-row__icon--before {
	margin-right: 1rem;
}
