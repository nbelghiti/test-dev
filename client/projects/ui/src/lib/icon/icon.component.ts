import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
