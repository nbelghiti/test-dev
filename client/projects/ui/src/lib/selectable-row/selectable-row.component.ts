import {
	Component,
	Input,
	OnDestroy,
	forwardRef
} from '@angular/core';
import { ControlValueAccessor, FormControl, NG_VALUE_ACCESSOR } from '@angular/forms';
import { Subject } from 'rxjs';
import { takeUntil } from 'rxjs/operators';

import { IconName } from '../icon/icon.types';

@Component({
	selector: 'ui-selectable-row',
	templateUrl: './selectable-row.component.html',
	styleUrls: ['./selectable-row.component.scss'],
	providers: [
		{
			provide: NG_VALUE_ACCESSOR,
			useExisting: forwardRef(() => SelectableRowComponent),
			multi: true,
		},
	],
})

export class SelectableRowComponent implements ControlValueAccessor, OnDestroy {
	@Input() public iconBefore?: string;
	@Input() public name: string;
	@Input() public value: string;

	public isChecked: FormControl = new FormControl(false);
	public IconName: typeof IconName = IconName;

	private destroyed$: Subject<boolean> = new Subject<boolean>();

	constructor() {
		this.isChecked.valueChanges
			.pipe(takeUntil(this.destroyed$))
			.subscribe((isChecked: boolean) => this.onChange(isChecked));
	}

	public ngOnDestroy(): void {
		this.destroyed$.next(true);
		this.destroyed$.complete();
	}

	public onChange: (_) => any = () => { };

	public onTouched = () => { };

	public writeValue(isChecked: boolean): void {
		if (isChecked !== this.isChecked.value) {
			this.isChecked.patchValue(isChecked);
		}
	}

	public registerOnChange(onChange: (_) => any): void {
		this.onChange = onChange;
	}

	public registerOnTouched(fn: () => void): void {
		this.onTouched = fn;
	}
}
