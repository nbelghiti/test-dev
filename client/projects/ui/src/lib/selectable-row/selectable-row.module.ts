import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';

import { IconModule } from '../icon/icon.module';

import { SelectableRowComponent } from './selectable-row.component';

@NgModule({
	imports: [
		CommonModule,
		RouterModule,
		ReactiveFormsModule,
		IconModule,
	],
	declarations: [
		SelectableRowComponent,
	],
	exports: [
		SelectableRowComponent,
	],
})
export class SelectableRowModule {}
