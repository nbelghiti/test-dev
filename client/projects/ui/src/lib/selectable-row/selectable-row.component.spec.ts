import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { RouterTestingModule } from '@angular/router/testing';

import { IconComponent } from '../icon/icon.component';

import { SelectableRowComponent } from './selectable-row.component';

describe('SelectableRowComponent', () => {
	let component: SelectableRowComponent;
	let fixture: ComponentFixture<SelectableRowComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			imports: [
				RouterTestingModule,
				ReactiveFormsModule
			],
			declarations: [
				SelectableRowComponent,
				IconComponent
			]
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(SelectableRowComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});

	describe('destroy', () => {
		beforeEach(() => {
			fixture = TestBed.createComponent(SelectableRowComponent);
			component = fixture.componentInstance;

			fixture.detectChanges();
		});

		it('should set a value to the destroyed$ Subject, destroying all subscriptions', async(done: jest.DoneCallback) => {
			const testSub = component['destroyed$']
				.subscribe((triggered: boolean) => {
					expect(triggered).toBe(true);

					testSub.unsubscribe();

					done();
				});

			component.ngOnDestroy();
		});
	});

	describe('writeValue', () => {
		beforeEach(() => {
			fixture = TestBed.createComponent(SelectableRowComponent);
			component = fixture.componentInstance;

			fixture.detectChanges();
		});

		it('should set the isChecked property', () => {
			expect(component.isChecked.value).toBe(false);

			component.writeValue(true);

			expect(component.isChecked.value).toBe(true);
		});
	});
});
