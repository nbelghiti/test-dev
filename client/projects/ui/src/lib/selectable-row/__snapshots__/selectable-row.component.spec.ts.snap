// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableRowComponent Should render correctly 1`] = `
<ui-selectable-row
  IconName={[Function Object]}
  destroyed$={[Function Subject]}
  isChecked={[Function FormControl]}
  onChange={[Function Function]}
  onTouched={[Function Function]}
>
  <label
    class="c-selectable-row"
  >
    <input
      class="c-selectable-row__checkbox ng-untouched ng-pristine ng-valid"
      ng-reflect-form="[object Object]"
      type="radio"
    />
    <div
      class="c-selectable-row__container"
    >
      <div
        class="c-selectable-row__content"
      >
        
      </div>
    </div>
  </label>
</ui-selectable-row>
`;
